### PR TITLE
[T4A3][W13-B4] Ang Wei Hao Desmond

### PR DESCRIPTION
**Pull request by awhdesmond**

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,53 +6,19 @@
...
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        super(address, isPrivate, EXAMPLE, MESSAGE_ADDRESS_CONSTRAINTS, ADDRESS_VALIDATION_REGEX);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:21]: :+1:  for moving validation to parent class, the right way

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:4]: Please include a header comment to indicate what this class/method does?

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+public class Contact {
+
+	private String example;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:7]: indentation is off here

File: src/seedu/addressbook/data/person/Contact.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+public class Contact {
+
+	private String example; 
+    private String validationRegex;
+	
+	protected String value;
+    protected boolean isPrivate;
+	
+    public Contact() {}
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:13]: why would you need an empty constructor here, considering that you don't use any empty constructor/super call in the child classes?

File: src/seedu/addressbook/data/person/Contact.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    public Contact(String value, boolean isPrivate, 
+    		       String example, String messageAddressConstraints, 
+    		       String validationRegex) throws IllegalValueException{
+        if (!isValid(value)) {
+            throw new IllegalValueException(messageAddressConstraints);
+        }
+        this.value = value;
+        this.isPrivate = isPrivate;
+        this.example = example;
+        this.validationRegex = validationRegex;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:29]: aren't you setting this after calling the `isValid` method?

File: src/seedu/addressbook/data/person/Contact.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    
+    public boolean isPrivate() {
+        return isPrivate;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:51]: aah! can you see the problem here? you can now potentially match `Address` with `Phone`; why?

**Comment by okkhoy** [comment]
@KnewYouWereTrouble 
Some comments added. Please ack & close the PR after reading comments.

**Comment by awhdesmond** [comment]
Acknowledged. Thanks for the feedback. Will improve the code and make amendments on my local.

**Pull request closed without merging**